allureServe/allureReport task failing after the second run if category.json is saved under test/resources

#### I'm submitting a ... 
  - [ x ] bug report

#### What is the current behavior?
Essential tasks are failing when running multiple times.

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem
 - include any categories.json file in the test/resource folder.
 - run tests
 - run `gradle allureReport`again (fine)
 - run `gradle allureReport` again 

```
> Task :copyCategories FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':copyCategories' (type 'CopyCategories').
  - In plugin 'io.qameta.allure.gradle.adapter.AllureAdapterBasePlugin' type 'io.qameta.allure.gradle.adapter.tasks.CopyCategories' property 'markerFile' is not writable because '/Users/vinogradov/git/bb-selenide/build/copy-categories/copyCategories' is not a file.
    
    Reason: Cannot write a file to a location pointing at a directory.
    
    Possible solutions:
      1. Configure 'markerFile' to point to a file, not a directory.
      2. Annotate 'markerFile' with @OutputDirectory instead of @OutputFiles.
    
    Please refer to https://docs.gradle.org/8.0/userguide/validation_problems.html#cannot_write_output for more details about this problem.


```

#### What is the expected behavior?

Tasks run multiple time without any problems.

#### Please tell us about your environment:

| Allure version | 2.21.1 |
| Gradle  | 7.4 and 8.0 |
| Gradle Allure Plugin | 2.11.2 |

--- a/allure-adapter-plugin/src/main/kotlin/io/qameta/allure/gradle/adapter/tasks/CopyCategories.kt
+++ b/allure-adapter-plugin/src/main/kotlin/io/qameta/allure/gradle/adapter/tasks/CopyCategories.kt
@@ -38,7 +38,7 @@ abstract class CopyCategories : DefaultTask() {
         set.map { it.resolve("categories.json") }
     }
 
-    @OutputFile
+    @OutputDirectory
     val markerFile = objects.directoryProperty()
         .convention(layout.buildDirectory.dir("copy-categories/$name"))
 

--- a/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/AllurePluginFeatureMatrixTest.kt
+++ b/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/AllurePluginFeatureMatrixTest.kt
@@ -46,6 +46,37 @@ class AllurePluginFeatureMatrixTest {
     }
 
     @Test
+    fun `allureReport can run twice with categories for both runtimes`() {
+        val projectDir = AllureRuntimeMatrixSupport.copyFixture(tempDir, "src/it/categories")
+        AllureRuntimeMatrixSupport.configureRuntime(projectDir, runtime, usesReportRuntime = true)
+
+        val firstRun = AllureRuntimeMatrixSupport.runner(projectDir)
+            .withArguments(AllureRuntimeMatrixSupport.commonArgs("allureReport"))
+            .build()
+
+        AllureRuntimeMatrixSupport.assertRuntimeTasks(firstRun, runtime)
+        assertThat(firstRun.task(":allureReport")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+        assertThat(projectDir.resolve("build/allure-results").listFiles().orEmpty().map { it.name })
+            .filteredOn { it.endsWith("categories.json") }
+            .hasSize(1)
+        assertThat(projectDir.resolve("build/reports/allure-report/allureReport"))
+            .isNotEmptyDirectory()
+
+        val secondRun = AllureRuntimeMatrixSupport.runner(projectDir)
+            .withArguments(AllureRuntimeMatrixSupport.commonArgs("allureReport"))
+            .build()
+
+        assertThat(secondRun.task(":allureReport")?.outcome)
+            .isIn(TaskOutcome.SUCCESS, TaskOutcome.UP_TO_DATE)
+        assertThat(projectDir.resolve("build/allure-results").listFiles().orEmpty().map { it.name })
+            .filteredOn { it.endsWith("categories.json") }
+            .hasSize(1)
+        assertThat(projectDir.resolve("build/reports/allure-report/allureReport"))
+            .isNotEmptyDirectory()
+    }
+
+    @Test
     fun `custom results dir is reused by report task for both runtimes`() {
         val projectDir = AllureRuntimeMatrixSupport.copyFixture(tempDir, "src/it/custom-results-dir")
         AllureRuntimeMatrixSupport.configureRuntime(projectDir, runtime, usesReportRuntime = true)

